z390test/src/main/groovy/org/z390/test/z390Test.groovy

```diff
@@ -0,0 +1,267 @@
+package org.z390.test
+
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+
+class z390Test {
+
+    var project_root = pathJoin('..')
+    String stdout = ""
+    String stderr = ""
+    var env = [:]
+    var fileData = [:]
+    var message = "z390Test - Test framework for z390 project"
+    int cmdTimeMs = 100000
+    boolean printOutput = false
+
+    File tempDir = null
+
+    static void main(String[] args) {
+        println new z390Test().message
+    }
+
+    z390Test() {
+        /**
+         * class constructor
+         */
+        if (System.getenv('Z390_PROJECT_ROOT')) {
+            this.project_root = new File(System.getenv('Z390_PROJECT_ROOT'))
+        }
+        if (System.getenv('Z390_PRINT_OUTPUT')) {
+            printOutput = true
+        }
+    }
+    @BeforeEach
+    void setUp() {
+        this.stdout = ""
+        this.stderr = ""
+        this.fileData = [:]
+    }
+    @AfterEach
+    void cleanUp() {
+        if (this.tempDir)
+            this.tempDir.deleteDir()
+    }
+
+    def createTempFile(String fileName, String fileContents, boolean returnExt=true) {
+        /**
+         * Utility method for creating temp option files used in tests
+         */
+        if (!tempDir) {
+            this.tempDir = File.createTempDir()
+            this.tempDir.deleteOnExit()
+        }
```

Comment:
The Groovy documentation I found at https://groovy-lang.org/gdk.html does not say where the temporary directory is created. Does this matter?

Reply:
No..  the location of the temp dir is OS dependant.. Linux will use /tmp, Windows uses a location in the user profile and MacOS has some special place that I can't recall at the moment.  Should not matter because the tempDir has a reference to it and can clean up after itself.